LEGO1/mxstring.cpp

```diff
@@ -79,3 +79,37 @@ const MxString &MxString::operator=(const char *param)
 
   return *this;
 }
+
+// Return type is intentionally just MxString, not MxString&.
+// This forces MSVC to add $ReturnUdt$ to the stack for 100% match.
```

Comment:
You need to return a copy, not a reference.
Otherwise you'd return a reference to an already-destroyed `MxString` on the stack.

Reply:
Does the final call to the MxString constructor at `0x100ae2a0` make that copy? (And does this actually leak memory for the tmp.m_data buffer used inside the function?)

Reply:
The caller allocates memory and passes this as an argument.
At the end, this method constructs a `MxString` in this memory.

A `MxString&` return type would just have returned a `MxString *` pointer, and not needed an extra argument.
Low-level, c++ references (`&` suffix) are the same as pointers.